manifests/agent/config.pp

```diff
@@ -17,7 +17,7 @@
     'usecacheonfailure': value => $::puppet::usecacheonfailure;
   }
   if !$::puppet::use_srv_records {
-    puppet::config::agent {
+    puppet::config::main {
```

Comment:
Should this be moved into `puppet::config` now, if it's no longer an agent-only setting? Other common settings are there, e.g. SRV server settings. This would allow it to be set when $agent = false, but the server is enabled.